sci-geosciences/josm/josm_bin-18531.recipe

```diff
@@ -0,0 +1,46 @@
+SUMMARY="An extensible editor for OpenStreetMap"
+DESCRIPTION="JOSM is the \"Java OpenStreetMap Editor\". It is a feature-rich editor \
+for the experienced OSM mapper. It requires some configuration efforts. But if you \
+intend to become a great OSM mapper, it's worth the time you need to get used to it."
+HOMEPAGE="https://josm.openstreetmap.de/"
+COPYRIGHT="2006-2022 Immanuel Scholz, Dirk Stöcker"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://josm.openstreetmap.de/download/josm-snapshot-$portVersion.jar#noarchive"
+CHECKSUM_SHA256="fdeb066cb6e2750eb4f9ab8c0b55b61b01fb579a9d4d5cdb94336cb24ba1723b"
+ADDITIONAL_FILES="josm.hvif"
+
+ARCHITECTURES="any"
+DISABLE_SOURCE_PACKAGE="true"
+
+PROVIDES="
+	josm_bin = $portVersion
+	app:josm = $portVersion
+	"
+REQUIRES="
+	haiku
+	java:environment == 8
+	"
+
+BUILD_PREREQUIRES="
+	"
+
+INSTALL()
+{
+	josmDir=$appsDir/JOSM
+	mkdir -p $josmDir/java
+
+	cp josm-snapshot-$portVersion.jar $josmDir/java/josm.jar
+
+	cat << EOF > $josmDir/josm.sh
+#!/bin/sh -l
+APPDIR="\$(dirname -- "\$(readlink -f -- "\${0}")" )"
+CLASSPATH="\${APPDIR}/java/josm.jar"
+\${JDK8_HOME}/bin/java -cp "\${CLASSPATH}" -Djosm.restart=true org.openstreetmap.josm.gui.MainApplication "\$@"
+EOF
+	chmod 755 $josmDir/josm.sh
+
+	addattr -t icon -f $portDir/additional-files/josm.hvif \
+	  "BEOS:ICON" "$josmDir/josm.sh"
```

Comment:
No chance on converting the hvif icon into an rdef file and use that one to add it to the binary?
Not sure it's needed to provide the svg icon (not checked but I guess it's part of the source)?

Reply:
> Not sure it's needed to provide the svg icon (not checked but I guess it's part of the source)?

Yes, svg icon is part of the upstream source repo and I've used it to convert into hvif icon.

Reply:
OK, then you can drop it in here, when I can I also use the svg icon to export it to rdef, doing so also enables you to add some mimetype info.